dispatcher/dispatcher.go

```diff
@@ -40,7 +42,14 @@ func Do(ctx context.Context, numWorkers, maxIter int, callback func()) {
 	wg.Wait()
 }
 
-func sanitizeNumWorkers(numWorkers int) int {
+// New returns a Dispatcher initialized with numWorker.
+func New(numWorker int) Dispatcher {
+	numWorker = sanitizeNumWorker(numWorker)
```

Comment:
At this point why not inline this function ?
```suggestion
	if numWorkers < 1 {
		numWorkers = 1
	}
```

Same for the 2 others?

Reply:
Since we are supposed to call `Validate()` on the config before declaring the dispatcher, couldn't we just get rid of all the sanitize methods because we can suppose that all configurations are valid?

Reply:
To me `dispatcher` should remain a completely independent package with its own logic, and making assumptions on its inputs based on how it's used outside is already a form of couplage. The external config validation is subject to change, and this package shouldn't depend on it. For that reason I believe it should have its own error handling.

As for the inlining, agreed!